code/__DEFINES/skills.dm

```diff
@@ -171,3 +40,85 @@
 #define SKILL_TASK_DIFFICULT	100
 #define SKILL_TASK_CHALLENGING	150
 #define SKILL_TASK_FORMIDABLE	200
+
+#define DEFAULT_SKILLSET /datum/skillset/pfc
+
+//D(irect) as in they don't perform the mind and skills datums existence sanity check. Use these only if their existence has been already checked.
+#define D_GET_SKILL_MOD(user, skill) (user.mind.cm_skills.skill_mods ? user.mind.cm_skills.skill_mods[skill] : SKILL_LEVEL_NONE)
+#define D_GET_SKILL(user, skill) (user.mind.cm_skills.omnighost ? SKILL_LEVEL_MASTER : CLAMP((user.mind.cm_skills.skillset ? user.mind.cm_skills.skillset[skill] : SKILL_LEVEL_NONE) + D_GET_SKILL_MOD(user, skill), SKILL_LEVEL_MIN, SKILL_LEVEL_MAX))
+#define D_HAS_SKILL_LEVEL(user, skill, level) (D_GET_SKILL(user, skill) >= level)
+#define D_GET_SKILL_MOD_ID(user, skill, id) (user.mind.cm_skills.skill_mod_ids && user.mind.cm_skills.skill_mod_ids[skill] ? user.mind.cm_skills.skill_mod_ids[skill[id]] : null)
+#define D_GET_SKILL_DIFF(user, skill, level) abs(level - D_GET_SKILL(user, skill))
+
+#define D_ADD_SKILL_MOD(user, skill, id, mod) \
```

Comment:
There doesn't seem to be any gain to putting these in as a DEFINE. Can this not just be a proc on /mob or even /living (whatever should have skills).

Reply:
Are you saying all these macros should be turned into procs or only the multi-line ones?

Reply:
@psykzz , procs are good if you need to redefine them, but unlike in other sane languages BYOND has an issue with proc overhead. Meaning we a) have a global limit on how many active procs there can be and b) they actually decrease performance.
Having defines/macros for stuff that needs to redefining is good and should be encouraged as long as it doesn't obliterate readability.